### PR TITLE
Implement alert list feature

### PR DESCRIPTION
**Pull request by haakonsf**
First PR for #1560. Further design and functionalities should be discussed in UX meeting

### CODE REVIEW

**Review by betaniat — COMMENTED** [review]
LGTM :) 
Screenshot 2024-08-21 at 10 23 18 AM

Alerts displayed, and queue mission button working as expected.

File: frontend/src/components/Contexts/AlertContext.tsx [hunk 1]
```diff
@@ -197,14 +250,23 @@ export const AlertProvider: FC<Props> = ({ children }) => {
...
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [newFailedMissions])
 
     useEffect(() => {
         const newBlockedRobotNames = enabledRobots
-            .filter((robot) => robot.status === RobotStatus.Blocked)
+            .filter(
+                (robot) =>
+                    robot.currentInstallation.installationCode.toLocaleLowerCase() ===
```
Comment (Eddasol) [inline frontend/src/components/Contexts/AlertContext.tsx:267]: Why is there added an installation filter here?

**Comment by Eddasol** [comment]
Need to clean up commit history

File: frontend/src/components/Contexts/AlertContext.tsx (hunk 1 above)
Comment (haakonsf) [reply frontend/src/components/Contexts/AlertContext.tsx]: Resolved when cleaning commit history

File: frontend/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -35,13 +45,41 @@ const StyledTopBarHeader = styled.div`
...
+        setIsAlertDialogOpen(true)
+    }
+
+    const onAlertClose = () => {
+        setIsAlertDialogOpen(false)
+    }
+
     return (
         <>
-            <StyledTopBar id={FrontPageSectionId.TopBar}>
```
Comment (Eddasol) [inline frontend/src/components/Header/Header.tsx:44]: Why is the id removed in this PR?

Comment (haakonsf) [reply frontend/src/components/Header/Header.tsx:44]: Reverted this

File: frontend/src/components/Alerts/AlertsListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                                <MissionRestartButton
+                                    mission={mission}
+                                    hasFailedTasks={missionHasFailedTasks}
+                                    smallButton={false}
+                                />
+                            )}
+                    </Right>
+                </StyledListItem>
+            </StyledListContainer>
+        </>
```
Comment (Eddasol) [inline frontend/src/components/Alerts/AlertsListItem.tsx:108]: The outermost <><> isnt necessary when its already wrapped in a style
```suggestion
            <StyledListContainer>
                <StyledListHeading>
                    <VerticalContent>
                        <StyledIcon name={icon} style={{ color: iconColor }} />
                        <Typography variant="h6">{alertTitle}</Typography>
                    </VerticalContent>
                </StyledListHeading>
                <StyledListItem>
                    <Typography variant="caption">{alertText}</Typography>
                    <Right>
                        {mission !== undefined &&
                            mission.tasks[0]?.type !== TaskType.ReturnHome &&
                            mission.tasks[0]?.type !== TaskType.Localization && (
                                <MissionRestartButton
                                    mission={mission}
                                    hasFailedTasks={missionHasFailedTasks}
                                    smallButton={false}
                                />
                            )}
                    </Right>
                </StyledListItem>
            </StyledListContainer>
```

File: frontend/src/components/Alerts/AlertsListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                {children}
+                <Button
+                    variant="ghost_icon"
+                    onClick={dismissAlert}
+                    style={{ marginTop: '0px', right: '25px', position: 'absolute' }}
+                >
+                    <Icon name={Icons.Clear}></Icon>
+                </Button>
+            </Horizontal>
+        </>
```
Comment (Eddasol) [inline frontend/src/components/Alerts/AlertsListItem.tsx:125]:
```suggestion
            <Horizontal>
                {children}
                <Button
                    variant="ghost_icon"
                    onClick={dismissAlert}
                    style={{ marginTop: '0px', right: '25px', position: 'absolute' }}
                >
                    <Icon name={Icons.Clear}></Icon>
                </Button>
            </Horizontal>
```

**Review by Eddasol — COMMENTED** [review]
Not a fan of how much duplicated code there is, but ok as a first iteration of the alert feature

**Review by Eddasol — APPROVED** [review]

**Pull request merged**